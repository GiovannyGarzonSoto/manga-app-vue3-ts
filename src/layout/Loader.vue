<template>
    <div class="loader">
        <p class="loader__title">Cargando...</p>
    </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
    name: 'loader',
    setup() {
        return {
        }
    }
})
</script>

<style lang="scss">
.loader{
    margin-top: var(--height-nav);
    position: absolute;
    inset: 0;
    background: rgb(13,13,13);
    z-index: 1;

    &__title{
        margin-top: 4rem;
        color: var(--color-small-impact);
        width: 100%;
        height: 8rem;
        background: linear-gradient(90deg, rgba(220,9,20,0) 0%, rgba(220,9,20,1) 50%, rgba(220,9,20,0) 100%);
        display: flex;
        font-weight: 800;
        justify-content: center;
        align-items: center;
        font-size: 2.4rem;
        letter-spacing: 1px;
        will-change: letter-spacing;
        animation-name: letterSpacingAnimation;
        animation-duration: 1s;
        animation-iteration-count: 1;
    }
    @keyframes letterSpacingAnimation {
        0%{
            letter-spacing: 24px;
        }
        100%{
            letter-spacing: 1px;
        }
    }
}
</style>