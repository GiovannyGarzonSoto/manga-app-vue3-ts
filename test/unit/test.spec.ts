import Loader from '../../src/layout/Loader.vue'
import {describe, expect, test} from 'vitest'
import { shallowMount } from '@vue/test-utils'

describe('setup test suit example', () => {
   test('should be match with snapshot', () => {
      const wrapper = shallowMount(Loader)
      expect(wrapper.html()).toMatchSnapshot()
   })
})