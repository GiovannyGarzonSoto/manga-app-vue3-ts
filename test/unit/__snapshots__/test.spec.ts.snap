// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`setup test suit example > should be match with snapshot 1`] = `
"<div class=\\"loader\\">
  <p class=\\"loader__title\\">Cargando...</p>
</div>"
`;
